kpi/views/v2/user.py

```diff
@@ -47,6 +51,28 @@ def list(self, request, *args, **kwargs):
             serializer = self.get_serializer(page, many=True)
             return self.get_paginated_response(serializer.data)
 
+    @action(
+        detail=True,
+        methods=['DELETE'],
+        renderer_classes=[renderers.JSONRenderer],
+    )
+    def delete(self, request, username, **kwargs):
+        if request.user.username != username:
+            raise exceptions.PermissionDenied
+
+        user = {'pk': request.user.pk, 'username': username}
+        try:
+            move_to_trash(
+                request.user, [user], config.ACCOUNT_TRASH_GRACE_PERIOD, 'user'
+            )
+        except TrashIntegrityError:
+            return Response({'error': t('User has already being deactivated')})
```

Comment:
I have changed the location of the endpoint to `/me/` instead of `/api/v2/users/<username>/` . 
Then, this error should never happen. I left a comment in the code. 
The endpoint returns an empty 204 on success.